packages/geo/src/lib/workspace/workspace-selector/workspace-selector.directive.ts

```diff
@@ -78,20 +83,54 @@ export class WorkspaceSelectorDirective implements OnInit, OnDestroy {
       return;
     }
     if (layer.dataSource instanceof WFSDataSource) {
-      return this.wfsWorkspaceService.createWorkspace(layer as VectorLayer, this.map);
+      const wfsWks = this.wfsWorkspaceService.createWorkspace(layer as VectorLayer, this.map);
+      this.computeTableTemplate(wfsWks);
+      return wfsWks;
     } else if (layer.dataSource instanceof WMSDataSource) {
       return this.wmsWorkspaceService.createWorkspace(layer as ImageLayer, this.map);
+    } else if (layer.dataSource instanceof FeatureDataSource && (layer as VectorLayer).exportable === true) {
+      const featureWks = this.featureWorkspaceService.createWorkspace(layer as VectorLayer, this.map);
+      this.computeTableTemplate(featureWks);
+      return featureWks;
     }
 
     return;
   }
 
+  private computeTableTemplate(workspace: Workspace) {
```

Comment:
This is better, thank you! Although I don't think the directive is the place this should be. If we were to activate a workspace from elsewhere, this wouldn't work. I don't think it's in the scope of the directive either. The WFSWorkspace and FeatureWorkspace could extend the activate method of their parent class to do just that.

There is another potential problem that I want to raise. What happens if the table is rendered before the columns are computed (since this is async)?

Reply:
moved into the original "createTableTemplate" into the wfs/feature workspace-service

Reply:
If you could take a look into this fix.